src/Halogen/HTML/Core.purs

```diff
@@ -57,6 +58,10 @@ instance bifunctorHTML :: Bifunctor HTML where
 instance functorHTML :: Functor (HTML p) where
   map = rmap
 
+renderWidget ∷ ∀ p q i j. (i → j) → (p → HTML q j) → HTML p i → HTML q j
+renderWidget f g (HTML vdom) =
```

Comment:
The function we use for constructing this below is `slot`, but here it's called `Widget`.

Reply:
Yeah, I wondered what to call it - since we still call the type var `p` for "placeholder" too, which is super duper old terminology. What do you think... `renderSlot`? `replaceSlot`?

Reply:
Is `Core.slot` used as primitive anywhere other than in `HTML.slot`? Maybe `Core.slot` should be `Core.widget` and then `renderWidget` is A-OK! 😄

Reply:
I like that, yeah.